enumerable_methods.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i])
+    end
+  end
+
+  def my_each_with_index
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i], i)
+    end
+  end
+
+  def my_select
+    instance = self
+    result = []
+    instance.my_each do |e|
+      result.push(e) if yield(e)
+    end
+    result
+  end
+
+  def my_all?
+    instance = self
+    result = true
+    instance.my_each do |e|
+      unless yield(e)
+        result = false
+        break
+      end
+    end
+    result
+  end
+
+  def my_any?
+    instance = self
+    result = false
+    instance.my_each do |e|
+      if yield(e)
+        result = true
+        break
+      end
+    end
+    result
+  end
+
+  def my_none?
+    instance = self
+    result = true
+    instance.my_each do |e|
+      if yield(e)
+        result = false
+        break
+      end
+    end
+    result
+  end
+
+  def my_count
+    instance = self
+    instance.length
+  end
+
+  def my_map
+    instance = self
+    result = []
+    instance.my_each do |e|
+      result.push(yield(e))
+    end
+    result
+  end
+
+  def my_inject
+    instance = self
+    total = instance.shift
+    instance.each do |e|
+      total = yield(total, e)
+    end
+    total
+  end
```

Comment:
- [ ] The method should be able to accept primitives, to_proc symbols (:+,:*, etc..) as arguments, yours raise an error when doing so :red_circle: 

example: 
```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_inject?(:+)
     #ArgumentError:
     #  wrong number of arguments (given 1, expected 0)
     # ./enumerable_methods.rb:27:in `my_inject?'
  ```